core/src/test/java/org/kohsuke/stapler/export/JSONDataWriterTest.java

```diff
@@ -0,0 +1,58 @@
+package org.kohsuke.stapler.export;
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.StringWriter;
+
+import static org.junit.Assert.assertEquals;
+
+public class JSONDataWriterTest {
+    private static <T> String serialize(T bean, Class<T> clazz) throws IOException {
+        StringWriter w = new StringWriter();
+        Model<T> model = new ModelBuilder().get(clazz);
+        model.writeTo(bean, Flavor.JSON.createDataWriter(bean, w));
+        return w.toString();
+    }
+
+    @ExportedBean public static class X {
+        @Exported public String a = "aval";
+        public String b = "bval";
+        @Exported public String getC() {return "cval";}
+        public String getD() {return "dval";}
+    }
+
+    @Test
+    public void testSimpleUsage() throws Exception {
+        assertEquals("{\"class\":\"X\",\"a\":\"aval\",\"c\":\"cval\"}",
```

Comment:
As of right now the unit tests are failing. Working on implementing https://github.com/stapler/stapler/pull/51#discussion_r33624844.